diff --git a/python/Plugins/Extensions/WeatherMSN/plugin.py b/python/Plugins/Extensions/WeatherMSN/plugin.py
index 0d4a3d5..e1c5f6f 100644
--- a/python/Plugins/Extensions/WeatherMSN/plugin.py
+++ b/python/Plugins/Extensions/WeatherMSN/plugin.py
@@ -667,13 +667,13 @@ class WeatherMSN(ConfigListScreen, Screen):
 		downloadPage(xmlfile, "/tmp/weathermsn1.xml").addCallback(self.downloadFinished).addErrback(self.downloadFailed)
 
 	def downloadFinished(self, result):
-		print "[WeatherMSN] Download finished"
+		print("[WeatherMSN] Download finished")
 		self.notdata = False
 		self.parse_weather_data()
 
 	def downloadFailed(self, result):
 		self.notdata = True
-		print "[WeatherMSN] Download failed!"
+		print("[WeatherMSN] Download failed!")
 
 	def get_weather_data(self):
 		if not os.path.exists("/tmp/weathermsn1.xml") or int((time.time() - os.stat("/tmp/weathermsn1.xml").st_mtime)/60) >= self.time_update or self.notdata:
@@ -2836,7 +2836,7 @@ def search_title(id):
 	try:
 		watchvideopage = urlopen(watchrequest)
 	except (URLError, HTTPException, socket.error) as err:
-		print "[Location] Error: Unable to retrieve page - Error code: ", str(err)
+		print("[Location] Error: Unable to retrieve page - Error code: ", str(err))
 	content = watchvideopage.read()
 	root = cet_fromstring(content)
 	search_results = []
