From b1ccc4e9e49a5567309fe4e4c14ff4bf4d97b0ed Mon Sep 17 00:00:00 2001
From: Hains <hainsvdbosch@ziggo.nl>
Date: Thu, 16 Apr 2020 20:35:08 +0200
Subject: [PATCH] Map python2 c api functions to python3 ones

---
 servicemp3/pythonmodule.cpp | 18 ++++++++++++++----
 servicemp3/servicemp3.h     |  2 ++
 2 files changed, 16 insertions(+), 4 deletions(-)

diff --git a/servicemp3/pythonmodule.cpp b/servicemp3/pythonmodule.cpp
index 8a4f619..c126fe1 100644
--- a/servicemp3/pythonmodule.cpp
+++ b/servicemp3/pythonmodule.cpp
@@ -1,13 +1,23 @@
 #include "servicemp3.h"
 
-
 static PyMethodDef servicemp3Methods[] =
 {
 	{NULL,NULL,0,NULL}
 };
 
-PyMODINIT_FUNC
-initservicemp3(void)
+static struct PyModuleDef servicemp3_moduledef = {
+	PyModuleDef_HEAD_INIT,
+	"servicemp3",         /* m_name */
+	"servicemp3",        /* m_doc */
+	-1,                  /* m_size */
+	servicemp3Methods,   /* m_methods */
+	NULL,                /* m_reload */
+	NULL,                /* m_traverse */
+	NULL,                /* m_clear */
+	NULL,                /* m_free */
+};
+
+PyMODINIT_FUNC PyInit_servicemp3(void)
 {
-	Py_InitModule("servicemp3", servicemp3Methods);
+	PyModule_Create(&servicemp3_moduledef);
 }
-- 
2.20.1

